rust-toolchain.toml

```diff
@@ -1,4 +1,7 @@
 [toolchain]
-channel = "1.75"
+channel = "1.77"
```

Comment:
Was there a specific reason for the update? I see that the MSRV is still at 1.75.0.

Reply:
This change is not required, but maybe could be worth updating the Rust version and MSRV. We can use a separate rust-toolchain.toml for Nix without modifying this one if needed.
What do you think?

Reply:
Thanks @ljoss17, for looking into that.
Actually, it doesn't break anything. Though, I lean towards keeping it consistent with the `sovereign-sdk-wip` and sticking to the project's MSRV, which is currently at 1.75.0. I use `cargo msrv` to determine the MSRV of the project. This way, if a dependency update causes us to break the current MSRV, we'll be aware of it.